### ISSUE TITLE
increase per user security/view security granularity

### ISSUE THREAD

**Issue by modem7** [BookStackApp/BookStack#2456]
**Describe the feature you'd like**
Currently we are able to either set books/shelves/etc to "all" or "own". But that means that ultimately, Bookstack is lacking a middle option where we can lock down books/shelves to specific users. 

E.g. Let's say I have a shelf called "Media watchlists", and in there, several books for different users. "Bob's watchlist", "Fred's watchlist" etc. 

Whilst I can set permissions to "own" in a security group such as (for example) "Media Watchlist users", people would still be able to see all books if I set everything to "all" in public/viewer security groups, which technically would make public/viewer groups less restrictive than internal groups. 

Basically, I would like to be able to hide/show books/shelves to different specific users rather than entire groups, whilst still allowing public/unregistered users to view public content that doesn't have specific permissions set. 

I might be missing something of course, if so, please let me know!

**Describe the benefits this feature would bring to BookStack users**
This would allow the ability to selectively hide/show books at a user level, rather than at a group level. 

Hopefully this makes sense!

**Comment by modem7** [comment]
Just to (try) to summarise: 

Bookshelf: Media Watchlists
Books: Fred's Watchlist, Bob's Watchlist

Public/Viewer permissions - when they go to bookshelf should not be able to see Media Watchlist books or shelves, but are able to see all books that do not have specific permissions such as "Media Watchlist". 

Media Watchlist permissions - They are able to see the bookshelf, but not only see all books that Public/Viewer can, but their books under Media Watchlist Bookshelf. So if Fred logged in, he'd see public books/shelves + media watchlist bookshelf + "Fred's Watchlist" book, but not "Bob's Watchlist" book

**Comment by ssddanbrown** [comment]
Thanks for the request @modem7.

This sounds like a combination of #410 and #1747 so I'm going to close this off to avoid overlapping issues. 
There's a "Permission System Review" step on the [roadmap](https://github.com/BookStackApp/BookStack#%EF%B8%8F-road-map) where these kind of things will be considered.
